### DISCUSSION TITLE
term_exists not detecting recently created terms while object cache is active

### DISCUSSION THREAD

**Discussion by seldimi** [rhubarbgroup/.github#23] [Support (ROC)] [answered]
Hello
Since WordPress 6.0, term_exists is using get_terms and caches functionality. Let's suppose we a foreach a loop where we need to check or create a taxonomy term on a loop. This will create 5 similar terms with name "test term". If you re-run the code, term_exists is running ok, so no more duplicates. If you delete all terms and modify for loop to 10, and run, it will create 10 instances of "test term"

That was not happening prior to 5.9 WordPress.

```
for ($i=0;$i<5;$i++) {
	create_term("test term","product_cat",[]);
}

function create_term($name,$taxonomy,$args  ) {
		$term = term_exists( $taxonomy_term, $taxonomy, $args['parent'] ?? 0 );

		if (0 !== $term && null !== $term) {
			$term_id = $term['term_id'];
		} else {
			$term = wp_insert_term($taxonomy_term, $taxonomy,$args);
			if (!is_wp_error($term)) {
				$term_id = $term['term_id'];
			} else {
				$term_id = null;
			}
		}
	}
```

Any ideas on how to fix that?

**Comment by seldimi** [comment]
To help up a bit, tried both solutions here but non worked.
https://make.wordpress.org/core/2022/04/28/taxonomy-performance-improvements-in-wordpress-6-0/

**Comment by tillkruss** [comment, marked as answer]
- You can disable query caching entirely in wp-core
- You can set a `maxttl` to let those cache keys expire
- You can not use `term_exists()` and run a raw database query yourself

**Comment by seldimi** [comment]
That was something that wasn't happening till 6.0, so if I got the same problem on multiple websites, more importers would do.
Can you advise on how to do the second bullet?

As a fallback, i m using renamed term_exists from WP 5.9 if REDIS is active, which is something I want to avoid.

**Reply by tillkruss** [reply]
This is a wp-core issue, nothing to do with this plugin.

As for the `maxttl`, please consult the `README`.

**Comment by seldimi** [comment]
I am trying to understand why it's core issue as it only happens when Redis is on?

**Reply by tillkruss** [reply]
WordPress is performing the cache operations. This plugin is just a link for WordPress to talk to Redis.
